tensorflow/lite/experimental/micro/kernels/cmsis-nn/reshape.cc

```diff
@@ -0,0 +1,100 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
```

Comment:
This file is not needed if it doesn't call arm_reshape_s8(). 

See: 
https://www.tensorflow.org/lite/microcontrollers/library#optimized_kernels
for more details. If you remove this function, the build system will fallback to use
'kernels/reshape.cc'
instead of 
'kernels/cmsis-nn/reshape.cc'

Reply:
Since reshape is just a memcpy, I would suggest not integrating this function from CMSIS-NN because it does not bring any performance gain.

Reply:
> This file (cmsis-nn/reshape.cc) is not needed if it doesn't call arm_reshape_s8().
> 
> See:
> https://www.tensorflow.org/lite/microcontrollers/library#optimized_kernels
> for more details. If you remove this function, the build system will fallback to use
> 'kernels/reshape.cc'
> instead of
> 'kernels/cmsis-nn/reshape.cc'

I have done that and also added the reference implementation fallback you suggested. 
Should I add the fallback for the ones you implemented? (_conv, depthwise_conv, and fully_connected_)

Reply:
> Since reshape is just a memcpy, I would suggest not integrating this function from CMSIS-NN because it does not bring any performance gain.


I thought in that direction too.
But thinking about it now, I think it makes more sense to use _arm_reshape_s8_ since ARM can change the implementation anytime irregardless of whether it's just a plain _memcpy_.

Reply:
> Since reshape is just a memcpy, I would suggest not integrating this function from CMSIS-NN because it does not bring any performance gain.

@lamarrr Considering that this function doesn't give a performance improvement, I think you can remove this file (kernels/cmsis-nn/reshape.cc).

Reply:
> > This file (cmsis-nn/reshape.cc) is not needed if it doesn't call arm_reshape_s8().
> > See:
> > https://www.tensorflow.org/lite/microcontrollers/library#optimized_kernels
> > for more details. If you remove this function, the build system will fallback to use
> > 'kernels/reshape.cc'
> > instead of
> > 'kernels/cmsis-nn/reshape.cc'
> 
> I have done that and also added the reference implementation fallback you suggested.
> Should I add the fallback for the ones you implemented? (_conv, depthwise_conv, and fully_connected_)

I wouldn't mind at all ;) Thanks!

Reply:
> > Since reshape is just a memcpy, I would suggest not integrating this function from CMSIS-NN because it does not bring any performance gain.
> 
> I thought in that direction too.
> But thinking about it now, I think it makes more sense to use _arm_reshape_s8_ since ARM can change the implementation anytime irregardless of whether it's just a plain _memcpy_.

You're exactly correct in the way you think. Me and @gmiodice represent ARM ;) we don't see optimization potential there currently - so we're discussing to remove it. @gmiodice , correct me if I'm wrong.

Reply:
Hi both, yes I confirm that since reshape does not bring any performance gain, we could remove that function from CMSIS-NN. Thanks for checking @freddan80 and thanks @lamarrr for your contribution ;)

Reply:
> > > Since reshape is just a memcpy, I would suggest not integrating this function from CMSIS-NN because it does not bring any performance gain.
> > 
> > 
> > I thought in that direction too.
> > But thinking about it now, I think it makes more sense to use _arm_reshape_s8_ since ARM can change the implementation anytime irregardless of whether it's just a plain _memcpy_.
> 
> You're exactly correct in the way you think. Me and @gmiodice represent ARM ;) we don't see optimization potential there currently - so we're discussing to remove it. @gmiodice , correct me if I'm wrong.

Great!
Alright, I'll be awaiting your conclusion.

Reply:
> Hi both, yes I confirm that since reshape does not bring any performance gain, we could remove that function from CMSIS-NN. Thanks for checking @freddan80 and thanks @lamarrr for your contribution ;)

My pleasure

Reply:
> So... I should remove the _cmsis-nn/reshape.cc_ source file?

yes

Reply:
> yes

ok. will do

Reply:
> > yes
> 
> ok. will do

Done